packages/minter-contracts/ligo/src/english_auction/english_auction_fa2.mligo

```diff
@@ -172,12 +204,25 @@ let resolve_auction(asset_id, storage : nat * storage) : return = begin
     let auction : auction = get_auction_data(asset_id, storage) in
     assert_msg (auction_ended(auction) , "Auction must have ended");
     assert_msg (Tezos.amount = 0mutez, "Amount must be 0mutez");
-
     let asset_transfers : operation list = transfer_tokens(auction.asset, Tezos.self_address, auction.highest_bidder) in
+
+#if !FEE
     let op_list : operation list = if first_bid(auction) then 
       asset_transfers else  
-      let return_bid : operation = fa2_fee_transfer(Tezos.self_address, auction.seller, auction.current_bid, storage.bid_currency) in 
-      (return_bid :: asset_transfers) in
+      let send_final_bid : operation = fa2_fee_transfer(Tezos.self_address, auction.seller, auction.current_bid, storage.bid_currency) in 
+      (send_final_bid :: asset_transfers) in
+#else 
+    
+    let op_list : operation list = if first_bid(auction) then 
+      asset_transfers else  
+      let fee : nat = percent_of_bid (storage.fee.fee_percent, auction.current_bid) in
+      let pay_fee : operation = fa2_fee_transfer(Tezos.self_address, storage.fee.fee_address, fee, storage.bid_currency) in
+      let final_price_minus_fee : nat =  (match (is_nat (auction.current_bid - fee)) with 
+        | Some adjusted_price -> adjusted_price 
+        | None -> (failwith "fee is larger than winning bid" : nat)) in
+      let send_final_bid_minus_fee : operation = fa2_fee_transfer(Tezos.self_address, auction.seller, final_price_minus_fee, storage.bid_currency) in 
```

Comment:
1. `fa2_fee_transfer` is a confusing function name since it is used not only to transfer fees but also auction payouts.

2. There could be one batched operation instead of two `pay_fee` and `send_final_bid_minus_fee`. Consider refactoring `fa2_fee_transfer` function to return a `transfer` record that can be later batched into a single FA2 `transfer` entry point call

Reply:
1. Changed to `single_fa2_transfer`
2. I refactored